includes/class-omise-capabilities.php

```diff
@@ -8,30 +8,63 @@ class Omise_Capabilities {
 	/**
 	 * @var self
 	 */
-	protected static $the_instance = null;
+	protected static $instance = null;
 
 	/**
 	 * @var \OmiseCapabilities
 	 */
 	protected $capabilities;
 
 	/**
+	 * Stores previously used public key to compare with the newly fetched public key
+	 * @var string
+	 */
+	protected $publicKey = null;
+
+	/**
+	 * Stores previously used secret key to compare with the newly fetched secret key
+	 * @var string
+	 */
+	protected $secretKey = null;
+
+	/**
+	 * @param string|null $pKey
+	 * @param string|null $sKey
+	 *
 	 * @return self  The instance of Omise_Capabilities
 	 */
-	public static function retrieve( $publickey = null, $secretkey = null ) {
-		if ( ! self::$the_instance ) {
-			try {
-				$capabilities = OmiseCapabilities::retrieve( $publickey , $secretkey );
-			} catch(\Exception $e) {
-				// suppressing error on the admin dashboard
-				return null;
+	public static function retrieve($pKey = null, $sKey = null)
+	{
+		$settings = Omise_Setting::instance();
+		$publicKey = !$pKey ? $settings->public_key() : $pKey;
+		$secretKey = !$sKey ? $settings->secret_key() : $sKey;
+
+		// Do not call capabilities API if keys are not present
+		if($publicKey === null || $secretKey === null) {
+			return null;
+		}
+
+		if(self::$instance) {
+			$keysNotChanged = self::$instance->publicKey === $publicKey && self::$instance->secretKey === $secretKey;
+
+			// if keys are same then no need to call capabilities API as t
```

Comment:
did you forget to complete comment?

Reply:
@ajzkk No idea what happened here. I have add more context. Thanks.